application/comit_node/src/swap_protocols/rfc003/bitcoin/actions.rs

```diff
@@ -13,6 +14,7 @@ pub struct SpendOutput {
     // Remember: One man's input is another man's output!
     pub output: PrimedInput,
     pub network: Network,
+    pub invalid_until: Option<Timestamp>,
```

Comment:
👃Putting `invalid_until` into the field of the action was not what I had in mind. I was thinking you would have to wrap it in a struct with an inner action and the timeout. Since you managed to get the API right I'm not going to block this PR here.

Reply:
I think we should do it right away in this PR. Nothing is really blocked and this PR introduces refund functionality so I think it should do it without any technical debt :)

Reply:
Starting to do the changes I am not sure I agree. `invalid_until` is part of the action, why would the internals be different than the API on this point?

For this is a parameter of the action, it is a parameter of the `SpendOutput`, the `nlocktime` needs to have a specific value related to `invalid_until` for this `SpendOutput` transaction to be as per Bitcoin consensus.

Reply:
@D4nte that's an interesting point and I'm tempted to agree with it but I think not being able to do execute an action until a time is something that potentially any action could have. Should Ethereum Deploy contract have a `invalid_until`? At first glance there is no particular reason it should, but then what if the constructor code checks the time and if it's too early something bad happens (note I have an idea I've shared with @bonomat that does exactly this).

Reply:
The work is done but to me it is an unnecessary to make it generic pre-emptively.